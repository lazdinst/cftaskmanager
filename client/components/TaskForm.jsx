import React from 'react';
import { Field, reduxForm } from 'redux-form';
import { Container, Input, Button, Card, Divider } from 'semantic-ui-react'

function renderField() {
  return (
    <Input
      transparent
    />
  );
}

function TaskForm({ handleSubmit, pristine, reset, submitting }) {
  return (
    <div style={{marginTop: '1.5rem'}}>
      <Container text>
        <Card.Group>
          <Card fluid>
            <Card.Content>
              <Button 
                basic
                icon='erase'
                floated='right'
                color='red'
                disabled={pristine || submitting} 
                onClick={reset}
              />
              <Button
                basic
                icon='checkmark'
                floated='right'
                color='green'
                type="submit"
                disabled={pristine || submitting}
                onClick={handleSubmit}
              />
              <Card.Header>
                <form onSubmit={handleSubmit}>
                  <div>
                    <Field
                      name='name'
                      component={Input}
                      className='ui transparent input'
                      type='text'
                      placeholder='Task Name'
                    />
                  </div>
                </form>
              </Card.Header>
              <Card.Meta>
            </Card.Meta>
            </Card.Content>
          </Card>
        </Card.Group>
      </Container>
    </div>
  );
};

export default reduxForm({
  form: 'crowdfund',
})(TaskForm);