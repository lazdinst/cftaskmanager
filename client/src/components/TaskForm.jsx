import React from 'react';
import { Field, reduxForm } from 'redux-form';
import { Container } from 'semantic-ui-react'

function TaskForm({ handleSubmit, pristine, reset, submitting }) {
  return (
    <Container text>
      <form onSubmit={handleSubmit}>
        <div>
          <Field
            name="name"
            component="input"
            type="text"
            placeholder="Task Name"
          />
        </div>
        <div>
          <button type="submit" disabled={pristine || submitting}>Submit</button>
          <button type="button" disabled={pristine || submitting} onClick={reset}>
            Clear
          </button>
        </div>
      </form>
    </Container>
  );
};

export default reduxForm({
  form: 'crowdfund',
})(TaskForm);