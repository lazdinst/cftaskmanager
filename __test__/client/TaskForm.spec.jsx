import React from 'react'
import { shallow, mount } from 'enzyme';
import TaskForm from '../../client/components/TaskForm.jsx'
import toJson from 'enzyme-to-json';

describe('<TaskForm />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<TaskForm />);
  });

  it('should render', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
