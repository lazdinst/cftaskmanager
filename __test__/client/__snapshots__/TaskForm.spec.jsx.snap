// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TaskForm /> should render 1`] = `
<Connect(Form(TaskForm))
  destroyOnUnmount={true}
  enableReinitialize={false}
  forceUnregisterOnUnmount={false}
  form="crowdfund"
  getFormState={[Function]}
  keepDirtyOnReinitialize={false}
  persistentSubmitErrors={false}
  pure={true}
  shouldAsyncValidate={[Function]}
  shouldError={[Function]}
  shouldValidate={[Function]}
  shouldWarn={[Function]}
  touchOnBlur={true}
  touchOnChange={false}
/>
`;
